logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/appVersioning/ObserveIfAppUpdateRequiredUseCase.kt

```diff
@@ -0,0 +1,107 @@
+package com.wire.kalium.logic.feature.appVersioning
+
+import com.wire.kalium.logic.configuration.server.ServerConfigRepository
+import com.wire.kalium.logic.data.auth.login.ProxyCredentials
+import com.wire.kalium.logic.di.MapperProvider
+import com.wire.kalium.logic.feature.UserSessionScopeProvider
+import com.wire.kalium.logic.feature.appVersioning.ObserveIfAppUpdateRequiredUseCaseImpl.Companion.CHECK_APP_VERSION_FREQUENCY_MS
+import com.wire.kalium.logic.feature.auth.AuthenticationScopeProvider
+import com.wire.kalium.logic.functional.getOrElse
+import com.wire.kalium.logic.functional.intervalFlow
+import com.wire.kalium.logic.functional.onFailure
+import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.logic.util.TimeParser
+import com.wire.kalium.logic.util.TimeParserImpl
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.coroutineContext
+
+/**
+ *
+ * Observes ServerConfigs and checks if App needs to be updated for each of it.
+ * Checking happens every [CHECK_APP_VERSION_FREQUENCY_MS] milliseconds, or when config was added into DB.
+ * @return Flow<Boolean> emits true if at least one ServerConfig requires app updating, false - otherwise.
+ *
+ */
+interface ObserveIfAppUpdateRequiredUseCase {
+    suspend operator fun invoke(currentAppVersion: Int): Flow<Boolean>
+}
+
+class ObserveIfAppUpdateRequiredUseCaseImpl internal constructor(
+    private val serverConfigRepository: ServerConfigRepository,
+    private val authenticationScopeProvider: AuthenticationScopeProvider,
+    private val userSessionScopeProvider: UserSessionScopeProvider,
+    private val timeParser: TimeParser = TimeParserImpl()
+) : ObserveIfAppUpdateRequiredUseCase {
+
+    override suspend fun invoke(currentAppVersion: Int): Flow<Boolean> {
```

Comment:
is it necessary to have an observable for is an update is required, afaik this is only done on app start and it will not be changed in run time

Reply:
it has to be `Observable` because: 
1. so when user logs in, we check "update requires" for the new ServerConfigs. 
2. case if app is opened for more than 24h in line ( 😜 ) we need to re-run the checking